### ISSUE TITLE
Annotates controller

### ISSUE THREAD

**Issue by heaven** [ctran/annotate_models#333]
I have model `User` and a `UsersController`, which always gets annotated. Even though I have it configured to add annotations at the end and in app/models. Very annoying!

```ruby
# == Schema Information
#
# Table name: users
#
#  id                     :integer          not null, primary key
#  email                  :string           default(""), not null
#  encrypted_password     :string           default(""), not null
#  reset_password_token   :string
#  reset_password_sent_at :datetime
#  remember_created_at    :datetime
#  sign_in_count          :integer          default(0), not null
#  current_sign_in_at     :datetime
#  last_sign_in_at        :datetime
#  current_sign_in_ip     :string
#  last_sign_in_ip        :string
#  confirmation_token     :string
#  confirmed_at           :datetime
#  confirmation_sent_at   :datetime
#  unconfirmed_email      :string
#  failed_attempts        :integer          default(0), not null
#  unlock_token           :string
#  locked_at              :datetime
#  created_at             :datetime         not null
#  updated_at             :datetime         not null
#
# Indexes
#
#  index_users_on_confirmation_token    (confirmation_token) UNIQUE
#  index_users_on_email                 (email) UNIQUE
#  index_users_on_reset_password_token  (reset_password_token) UNIQUE
#  index_users_on_unlock_token          (unlock_token) UNIQUE
#

class UsersController < ApplicationController; end
```

It also annotates the spec for this controller.

**Comment by olhor** [comment]
Confirmed, same problem. All controllers are annotated with misleading console message saying eg. ```Annotated (1): app/models/model_name.rb```
instead, controller is annotated.

**Comment by FME-Daniel** [comment]
https://github.com/ctran/annotate_models/releases/tag/v2.7.0

Controllers and helpers are annotated now

**Comment by heaven** [comment]
I have more than 25 projects and don't know any where I would need that feature. It seems I can skip controllers by using `exclude_controllers = false`, but it keeps annotating controller specs.

**Comment by FME-Daniel** [comment]
`exclude_controllers = false` means `do not exclude controllers`. You want to set that to true.

**Comment by heaven** [comment]
Whoops, sorry, that's what I meant :) Yes, I do use true for this.

**Comment by cschulte22** [comment]
I have to agree with @heaven on this one, I don't know any reason why I'd want the annotations in the controllers (or the helpers for that matter).  Seems like it should be off by default unless I want to go through the trouble of making a config file and specifically including them.

**Comment by ctran** [comment]
I have a plan in mind to bring back the default behaviors.  Stay tuned :)

**Comment by heaven** [comment]
If to annotate controllers it could make sense to put routes, related to each specific controller.

**Comment by olhor** [comment]
Thumbs up for bringing back default behaviors.

**Comment by cschulte22** [comment]
Just for reference for anyone else that views this issue, I was able to get the default behavior back by making a config file per the Readme (`rails g annotate:install`), then editing that file and making the following items `true` in the `auto_annotate_models.rake` file.

```
      'exclude_scaffolds'       => 'true',
      'exclude_controllers'     => 'true',
      'exclude_helpers'         => 'true',
```

Note that this will run the annotations automatically after migrations (not default behavior), so you need to set `'skip_on_db_migrate'      => 'true',` if you don't want that either.

@ctran I'm not sure what you have in mind, but with all this additional functionality maybe it makes sense to just always generate a well commented configuration file when the gem is installed.

**Comment by bodrovis** [comment]
@cschulte22 :thumbsup:

**Comment by theotherzach** [comment]
I just found annotations in my controller spec after turning off annotations in my controller. (Annotations in my controller!?!?!?! Really? It's your project and I thank you for your work and obviously take it in any direction you want, but really?) 

I skimmed the thread and didn't see any resolution to this, did I miss a way to turn off controller spec annotation?

**Comment by bruschill** [comment]
I'm having the same issue as @theotherzach. Controller specs are getting annotated even though I have controller annotations excluded.

**Comment by zbeat** [comment]
I'm having the same problem: I only want annotations for actual **model** files (i.e. `app/models/*.rb`). I originally thought the [README](https://github.com/ctran/annotate_models/blob/develop/README.rdoc) had my answer:
> To annotate just your models:
> 
> `annotate --exclude tests,fixtures,factories,serializers`

So I ran that on a specific model:
`$ annotate -iks --exclude tests,fixtures,factories,serializers app/models/user.rb`

But it still annotates the associated **controller**, as well as various **spec** files:
```
$ git status
...
	modified:   app/controllers/users_controller.rb
	modified:   app/models/user.rb
	modified:   spec/controllers/users_controller_spec.rb
	modified:   spec/requests/users_spec.rb
	modified:   spec/routing/users_routing_spec.rb
```

So I added controllers to list of exclusions:
`$ annotate -iks --exclude tests,fixtures,factories,serializers,controllers app/models/user.rb`

That stopped it from changing the controller itself, but the specs are still a problem:
```
$ git status
...
	modified:   app/models/user.rb
	modified:   spec/controllers/users_controller_spec.rb
	modified:   spec/requests/users_spec.rb
	modified:   spec/routing/users_routing_spec.rb
```

It looks like `--exclude tests` only skips the proper model spec (i.e. `spec/models/user_spec.rb`), but not the other tests associated w/ this model.   

While not the end of the world, it's super annoying! :persevere: What am I missing? This doesn't seem how I would expect/hope the gem to work.

Of course I can now just add the following step to my process:
```
$ git checkout spec/
$ git status
...
	modified:   app/models/user.rb
```

**Comment by heaven** [comment]
I am not getting the users controller annotated with this config:
```ruby
# NOTE: only doing this in development as some production environments (Heroku)
# NOTE: are sensitive to local FS writes, and besides -- it's just not proper
# NOTE: to have a dev-mode tool do its thing in production.
if Rails.env.development?
  task :set_annotation_options do
    # You can override any of these by setting an environment variable of the
    # same name.
    Annotate.set_defaults(
      'routes'                  => 'false',
      'position_in_routes'      => 'after',
      'position_in_class'       => 'after',
      'position_in_test'        => 'after',
      'position_in_fixture'     => 'after',
      'position_in_factory'     => 'after',
      'position_in_serializer'  => 'after',
      'show_foreign_keys'       => 'true',
      'show_indexes'            => 'true',
      'simple_indexes'          => 'false',
      'model_dir'               => 'app/models',
      'root_dir'                => '',
      'include_version'         => 'false',
      'require'                 => '',
      'exclude_tests'           => 'false',
      'exclude_fixtures'        => 'false',
      'exclude_factories'       => 'false',
      'exclude_serializers'     => 'false',
      'exclude_scaffolds'       => 'false',
      'exclude_controllers'     => 'true',
      'exclude_helpers'         => 'true',
      'ignore_model_sub_dir'    => 'false',
      'ignore_columns'          => nil,
      'ignore_unknown_models'   => 'false',
      'hide_limit_column_types' => 'integer,boolean',
      'skip_on_db_migrate'      => 'false',
      'format_bare'             => 'true',
      'format_rdoc'             => 'false',
      'format_markdown'         => 'false',
      'sort'                    => 'false',
      'force'                   => 'false',
      'trace'                   => 'false',
      'wrapper_open'            => nil,
      'wrapper_close'           => nil,
    )
  end

  Annotate.load_tasks
end
```

Using this rev: 56abbaa3eb5e41a3b236f782c0bb2a49dc3d36f1

**Comment by ZombieHarvester** [comment]
```
'exclude_controllers'     => 'true',
'exclude_tests'           => 'false',
'position_in_test'        => 'after',
```

it still annotates controller tests, adds schema info above (while in model tests schema is below respecting the setting ```'position_in_test'        => 'after'```) and in console the output is ```Annotated (4): app/models/account.rb...``` however in reality ```test/controllers/accounts_controller_test.rb``` is getting annotated

```Annotate Models``` is an awesome gem and many thanks to the author, but this behaviour is super annoying

**Comment by ctran** [comment]
I'll try to fix this up this weekend, stay tuned.

**Comment by ctran** [comment]
See #370.  This will be merged in 2.8.

**Comment by keonjeo** [comment]
I also have this same issue. 
-----------------------------------------------------
```'exclude_scaffolds' => 'true'``` , It works for me.
